pkg/cmd/pulumi/stack_init.go

```diff
@@ -74,119 +69,145 @@ func newStackInitCmd() *cobra.Command {
 			ctx := commandContext()
-			opts := display.Options{
-				Color: cmdutil.GetGlobalColorization(),
-			}
-
-			// Try to read the current project
-			project, _, err := readProject()
-			if err != nil && !errors.Is(err, workspace.ErrProjectNotFound) {
-				return err
-			}
-
-			b, err := currentBackend(ctx, project, opts)
-			if err != nil {
-				return err
-			}
-
-			if len(args) > 0 {
-				if stackName != "" {
-					return errors.New("only one of --stack or argument stack name may be specified, not both")
-				}
-
-				stackName = args[0]
-			}
-
-			// Validate secrets provider type
-			if err := validateSecretsProvider(secretsProvider); err != nil {
-				return err
-			}
-
-			if stackName == "" && cmdutil.Interactive() {
-				if b.SupportsOrganizations() {
-					fmt.Print("Please enter your desired stack name.\n" +
-						"To create a stack in an organization, " +
-						"use the format <org-name>/<stack-name> (e.g. `acmecorp/dev`).\n")
-				}
-
-				name, nameErr := promptForValue(false, "stack name", "dev", false, b.ValidateStackName, opts)
-				if nameErr != nil {
-					return nameErr
-				}
-				stackName = name
-			}
-
-			if stackName == "" {
-				return errors.New("missing stack name")
-			}
-
-			if err := b.ValidateStackName(stackName); err != nil {
-				return err
-			}
-
-			stackRef, err := b.ParseStackReference(stackName)
-			if err != nil {
-				return err
-			}
-
-			proj, root, projectErr := readProject()
-			if projectErr != nil && !errors.Is(projectErr, workspace.ErrProjectNotFound) {
-				return projectErr
-			}
-
-			// Backend-specific config options. Currently only applicable to the HTTP backend.
-			createOpts, err := validateCreateStackOpts(stackName, b, teams)
-			if err != nil {
-				return err
-			}
-
-			newStack, err := createStack(ctx, b, stackRef, root, createOpts, !noSelect, secretsProvider)
-			if err != nil {
-				return err
-			}
-
-			if stackToCopy != "" {
-				if projectErr != nil {
-					return projectErr
-				}
-
-				// load the old stack and its project
-				copyStack, err := requireStack(ctx, stackToCopy, stackLoadOnly, opts)
-				if err != nil {
-					return err
-				}
-				copyProjectStack, err := loadProjectStack(proj, copyStack)
-				if err != nil {
-					return err
-				}
-
-				// get the project for the newly created stack
-				newProjectStack, err := loadProjectStack(proj, newStack)
-				if err != nil {
-					return err
-				}
-
-				// copy the config from the old to the new
-				return copyEntireConfigMap(copyStack, copyProjectStack, newStack, newProjectStack)
-			}
-
-			return nil
+			return sicmd.Run(ctx, args)
 		}),
 	}
 	cmd.PersistentFlags().StringVarP(
-		&stackName, "stack", "s", "", "The name of the stack to create")
+		&sicmd.stackName, "stack", "s", "", "The name of the stack to create")
 	cmd.PersistentFlags().StringVar(
-		&secretsProvider, "secrets-provider", "default", possibleSecretsProviderChoices)
```

Comment:
Now defaults to `""`? I think that's fine pretty sure all the checks for "default" also check for "", just checking this was intended.